src/main/java/com/puppycrawl/tools/checkstyle/Main.java

```diff
@@ -597,32 +622,43 @@ private static void printUsage() {
         final HelpFormatter formatter = new HelpFormatter();
         formatter.setWidth(HELP_WIDTH);
         formatter.printHelp(String.format("java %s [options] -c <config.xml> file...",
-                Main.class.getName()), buildOptions());
+                Main.class.getName()), buildOptionsExcludeHidden());
     }
 
     /**
      * Builds and returns list of parameters supported by cli Checkstyle.
      * @return available options
      */
     private static Options buildOptions() {
+        final Options options = buildOptionsExcludeHidden();
+        options.addOption(OPTION_CAPITAL_M_NAME, OPTION_THREADS_NUMBER_COUNT_NAME, true,
+                "The number of threads (must be greater than zero)");
+        return options;
+    }
+
+    /**
+     * Builds and returns list of parameters supported by cli Checkstyle, excluding hidden options.
+     * @return available options (excluding hidden options)
+     */
+    private static Options buildOptionsExcludeHidden() {
         final Options options = new Options();
         options.addOption(OPTION_C_NAME, true, "Sets the check configuration file to use.");
         options.addOption(OPTION_O_NAME, true, "Sets the output file. Defaults to stdout");
         options.addOption(OPTION_P_NAME, true, "Loads the properties file");
         options.addOption(OPTION_F_NAME, true, String.format(
-                "Sets the output format. (%s|%s). Defaults to %s",
-                PLAIN_FORMAT_NAME, XML_FORMAT_NAME, PLAIN_FORMAT_NAME));
+            "Sets the output format. (%s|%s). Defaults to %s",
+            PLAIN_FORMAT_NAME, XML_FORMAT_NAME, PLAIN_FORMAT_NAME));
```

Comment:
Go through and undo superfluous changes like these. We need to make PR as concise and easy to review.